### PR TITLE
Cross-link the perf review document

### PR DESCRIPTION
**Pull request by agnivade**
The developers document is a more prominent place
where lots of contributors hit.

We copy some of the material from the doc to improve
visibilty and point to the rest of the doc.

### CODE REVIEW

**Review by streamer45 — APPROVED** [review]
Thanks this is great :clap:

File: site/content/contribute/server/style-guide.md [hunk 1]
```diff
@@ -257,6 +257,22 @@ func (worker *Worker) Run() {
 	..
 ```
 
+### Performance sensitive areas
+
+Any PR that can potentially have a performance impact on the `mattermost-server` codebase is encouraged to have a performance review. For more information, please see this [link](https://docs.google.com/document/d/1Uzt3XHyKhDKipkuCmESkHoPio7vz7VYS4N_5_9ffgNU/edit). Following is a brief list of indicators that are good items to undergo a performance review:
+
+- New features that might require benchmarks and/or are missing load-test coverage.
+- PRs touching performance critical parts of the codebase (e.g. `Hub`/`WebConn`).
```
Comment (cwarnermm) [inline site/content/contribute/server/style-guide.md:265]:
```suggestion
- PRs touching performance of critical parts of the codebase (e.g. `Hub`/`WebConn`).
```

File: site/content/contribute/server/style-guide.md [hunk 2]
```diff
@@ -257,6 +257,22 @@ func (worker *Worker) Run() {
...
+- PRs adding/updating SQL queries.
```
Comment (cwarnermm) [inline site/content/contribute/server/style-guide.md:266]:
```suggestion
- PRs adding or updating SQL queries.
```

File: site/content/contribute/server/style-guide.md [hunk 3]
```diff
@@ -257,6 +257,22 @@ func (worker *Worker) Run() {
...
+- Creating goroutines.
+- Doing possibly expensive allocations: `bytes.Buffer` and `[]byte`:
```
Comment (cwarnermm) [inline site/content/contribute/server/style-guide.md:268]:
```suggestion
- Doing potentially expensive allocations: `bytes.Buffer` and `[]byte`:
```

File: site/content/contribute/server/style-guide.md [hunk 4]
```diff
@@ -257,6 +257,22 @@ func (worker *Worker) Run() {
...
+	- Use of `Buffer.Grow`, `Buffer.ReadFrom`, `ioutil.ReadAll`.
+	- Creating big slices and maps without capacity when size is known in advance.
+- Recursion, unbounded and deeply nested `for` loops.
```
Comment (cwarnermm) [inline site/content/contribute/server/style-guide.md:271]:
```suggestion
- Recursion, unbounded, and deeply nested `for` loops.
```

File: site/content/contribute/server/style-guide.md [hunk 5]
```diff
@@ -257,6 +257,22 @@ func (worker *Worker) Run() {
...
+- Use of locks and/or other synchronization primitives.
+- Regular expressions, especially creating `regexp.MustCompile` dynamically every time.
```
Comment (cwarnermm) [inline site/content/contribute/server/style-guide.md:273]:
```suggestion
- Regular expressions, especially when creating `regexp.MustCompile` dynamically every time.
```

File: site/content/contribute/server/style-guide.md [hunk 6]
```diff
@@ -257,6 +257,22 @@ func (worker *Worker) Run() {
 	..
 ```
 
+### Performance sensitive areas
+
+Any PR that can potentially have a performance impact on the `mattermost-server` codebase is encouraged to have a performance review. For more information, please see this [link](https://docs.google.com/document/d/1Uzt3XHyKhDKipkuCmESkHoPio7vz7VYS4N_5_9ffgNU/edit). Following is a brief list of indicators that are good items to undergo a performance review:
```
Comment (cwarnermm) [inline site/content/contribute/server/style-guide.md:262]:
```suggestion
Any PR that can potentially have a performance impact on the `mattermost-server` codebase is encouraged to have a performance review. For more information, please see this [link](https://docs.google.com/document/d/1Uzt3XHyKhDKipkuCmESkHoPio7vz7VYS4N_5_9ffgNU/edit). The following is a brief list of indicators that should to undergo a performance review:
```

**Review by cwarnermm — APPROVED** [review]
This is a great addition to the dev docs. I've provided non-blocking suggestions inline.

File: site/content/contribute/server/style-guide.md (hunk 4 above)
Comment (agnivade) [reply site/content/contribute/server/style-guide.md:271]: Always love oxford commas ;)

**Comment by agnivade** [comment]
Thank you @cwarnermm !

**Pull request merged**